internal/ringbuffer.go

```diff
@@ -139,3 +135,66 @@ func (rp *RingbufferProxy) checkCounts(minCount int32, maxCount int32) (err erro
 	}
 	return
 }
+
+type LazyReadResultSet struct {
+	readCount            int32
+	items                []interface{}
+	itemSequences        []int64
+	serializationService *SerializationService
+}
+
+const sequenceUnavailable int64 = -1
+
+func NewLazyReadResultSet(readCount int32, itemsData []*Data, itemSeqs []int64, ss *SerializationService) (rs *LazyReadResultSet) {
+	rs = &LazyReadResultSet{readCount: readCount, itemSequences: itemSeqs, serializationService: ss}
+	rs.items = rs.convertToInterfaceSlice(itemsData)
```

Comment:
In the corresponding codec if we created `[]interface{}` instead of `[]*Data` we wouldnt need this.  We iterate the same slice twice because of that. 

Doing that for only that codec is not trivial unless we do it manually. 
So I am not sure which is better.

Reply:
The codecs are designed to be code generated, in java you can not even edit them. Let's keep it as `[]*Data`